File: OrchidCore/src/main/java/com/eden/orchid/impl/themes/components/LicenseComponent.java
package com.eden.orchid.impl.themes.components;

import com.eden.orchid.api.OrchidContext;
import com.eden.orchid.api.resources.resource.OrchidResource;
import com.eden.orchid.api.theme.components.OrchidComponent;

import javax.inject.Inject;

public final class LicenseComponent extends OrchidComponent {

    @Inject
    public LicenseComponent(OrchidContext context) {
        super(context, "license", 40);
    }

    public String getContent() {
        OrchidResource licenseResource = context.findClosestFile("license");
        if (licenseResource != null) {
            return licenseResource.compileContent(this);
        }
        return null;
    }
}

File: OrchidCore/src/main/kotlin/com/eden/orchid/impl/themes/components/LicenseComponent.kt
package com.eden.orchid.impl.themes.components

import com.eden.orchid.api.options.annotations.Description
import com.eden.orchid.api.resources.resourcesource.LocalResourceSource
import com.eden.orchid.api.theme.components.OrchidComponent
import com.eden.orchid.utilities.SuppressedWarnings

@Description(value = "Locate and display your project's License file.", name = "License")
@Suppress(SuppressedWarnings.DEPRECATION)
class LicenseComponent : OrchidComponent("license") {
    val content: String?
        get() = context
            .getFlexibleResourceSource(LocalResourceSource, null)
            .findClosestFile(context, "license")
            ?.compileContent(context, this)
}
